test/jdk/javax/swing/JButton/HtmlButtonWithTextAndIcon.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+
+/* @test
+ * @bug 8302173
+ * @requires (os.family == "mac")
+ * @summary Tests Aqua button icon positioning with HTML text
+ * @run main HtmlButtonWithTextAndIcon
+ */
+
+import java.awt.Color;
+import java.awt.Component;
+import java.awt.Dimension;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+
+import javax.imageio.ImageIO;
+import javax.swing.Icon;
+import javax.swing.JButton;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+
+import static java.awt.image.BufferedImage.TYPE_INT_ARGB;
+
+public class HtmlButtonWithTextAndIcon {
+    private static Path testDir;
+    private static JButton button;
+    private static BufferedImage image;
+    private static final int OFFSET = 1;
+    private static final int BUTTON_WIDTH = 59;
+    private static final int BUTTON_HEIGHT = 28;
+    private static final int ICON_WIDTH = 16;
```

Comment:
To make testing easier with different icon & button sizes, it is better to have button width and height as a factor of icon size (1.5 times or greater) as mentioned here [JDK-8015854](https://bugs.openjdk.org/browse/JDK-8015854).

Reply:
I had the same thought, but decided to use these values for width and height due to wanting to match the provided test in the issue. This was to best mimic the overlapping displayed in the screenshots, and using these fixed values seems like the best way for me to replicate the overlapping. These are the width & height values of the button in the original test provided so I used the same icon size, text, and button dimensions.